nc/source/component/vulkan/MeshRenderer.cpp

```diff
@@ -9,30 +10,42 @@ namespace nc::vulkan
 {
     MeshRenderer::MeshRenderer(Entity entity, std::string meshUid, nc::graphics::vulkan::Material material, nc::graphics::vulkan::TechniqueType techniqueType)
     : ComponentBase{entity},
-      m_meshUid{std::move(meshUid)},
+      m_mesh{nc::graphics::vulkan::ResourceManager::GetMeshAccessor(meshUid)},
       m_material{std::move(material)},
-      m_techniqueType{techniqueType}
+      m_techniqueType{techniqueType},
+      m_objectData{}
     {
+        m_objectData.baseColorIndex = nc::graphics::vulkan::ResourceManager::GetTextureAccessor(m_material.baseColor); // Todo: Make this more generic for materials;
+        m_objectData.normalIndex = nc::graphics::vulkan::ResourceManager::GetTextureAccessor(m_material.normal); // Todo: Make this more generic for materials;
+        m_objectData.roughnessIndex = nc::graphics::vulkan::ResourceManager::GetTextureAccessor(m_material.roughness); // Todo: Make this more generic for materials;
+        m_objectData.isInitialized = true;
     }
 
-    Transform* MeshRenderer::GetTransform()
+    const nc::graphics::vulkan::Mesh& MeshRenderer::GetMesh() const
     {
-        return ActiveRegistry()->Get<Transform>(GetParentEntity());
+        return m_mesh;
     }
 
-    const std::string& MeshRenderer::GetMeshUid() const
+    nc::graphics::vulkan::TechniqueType MeshRenderer::GetTechniqueType() const
     {
-        return m_meshUid;
+        return m_techniqueType;
     }
 
     nc::graphics::vulkan::Material& MeshRenderer::GetMaterial()
     {
         return m_material;
     }
 
-    nc::graphics::vulkan::TechniqueType MeshRenderer::GetTechniqueType() const
+    void MeshRenderer::Update(const  DirectX::FXMMATRIX& viewMatrix, const  DirectX::FXMMATRIX& projectionMatrix)
```

Comment:
I eventually want to get rid of ActiveRegistry(). The more places it is used, the harder it becomes, so can you pass the registry to this method? The system that calls this already has it.